core/lib/Tab.ts

```diff
@@ -817,7 +817,10 @@ export default class Tab
   private async waitForReady(): Promise<void> {
     await this.mainFrameEnvironment.isReady;
     if (this.session.options?.blockedResourceTypes) {
```

Comment:
I know this already got pulled in, but I think you want to also check if blocked resource urls was provided. Otherwise this feature only works when there are blocked resource _types_ as well.